### PR TITLE
docs: Update EnforcedTargetLimit description

### PR DESCRIPTION
**Pull request by bison**
The EnforcedTargetLimit configuration option was added in #3571, but
it seems where the docs say "if TargetLimit is higher that value will
be taken instead", that should instead read "if TargetLimit is lower."

```release-note:none
No release notes.
```

### CODE REVIEW

**Comment by bison** [comment]
I think the doc for this is incorrect. The actual logic is a bit more complex, i.e. checking if either value is zero, but it seems it should read that the individual TargetLimit will be preferred if it is *lower* not *higher*. Let me know if I'm misreading.

File: Documentation/api.md [hunk 1]
```diff
@@ -653,7 +653,7 @@ PrometheusSpec is a specification of the desired behavior of the Prometheus clus
...
 | allowOverlappingBlocks | AllowOverlappingBlocks enables vertical compaction and vertical query merge in Prometheus. This is still experimental in Prometheus so it may change in any upcoming release. | bool | false |
-| enforcedTargetLimit | EnforcedTargetLimit defines a global limit on the number of scraped targets. This overrides any TargetLimit set per ServiceMonitor or/and PodMonitor. It is meant to be used by admins to enforce the TargetLimit to keep overall number of targets under the desired limit. Note that if TargetLimit is higher that value will be taken instead. | *uint64 | false |
+| enforcedTargetLimit | EnforcedTargetLimit defines a global limit on the number of scraped targets. This overrides any TargetLimit set per ServiceMonitor or/and PodMonitor. It is meant to be used by admins to enforce the TargetLimit to keep overall number of targets under the desired limit. Note that if TargetLimit is lower that value will be taken instead. | *uint64 | false |
```
Comment (lilic) [inline Documentation/api.md:656]:
```suggestion
| enforcedTargetLimit | EnforcedTargetLimit defines a global limit on the number of scraped targets. This overrides any TargetLimit set per ServiceMonitor or/and PodMonitor. It is meant to be used by admins to enforce the TargetLimit to keep the overall number of targets under the desired limit. Note that if TargetLimit is lower, that value will be taken instead. | *uint64 | false |
```

Lets then add a note that if the enforcedTargetLimit is 0, this means no limit is set. This is due to how Prometheus target_limit works https://prometheus.io/docs/prometheus/latest/configuration/configuration/

**Review by lilic — APPROVED** [review]
One suggestion otherwise lgtm, thanks!

File: Documentation/api.md (hunk 1 above)
Comment (bison) [reply Documentation/api.md]: Updated. Let me know if this is better.

Comment (bison) [reply Documentation/api.md]: Ah, sorry I missed your other suggested change. I can update that as well.

**Review by lilic — APPROVED** [review]
/lgtm

Thanks!

File: Documentation/api.md (hunk 1 above)
Comment (bison) [reply Documentation/api.md]: Done.

**Pull request merged**